__tests__/app.test.js

```diff
@@ -36,32 +37,83 @@ describe("GET /api/topics", () => {
-              topic: expect.any(String),
-              created_at: expect.any(String),
-              votes: expect.any(Number),
-              article_img_url: expect.any(String),
-            });
-          });
+  test("200: return an article object with the correct properties based on the article id", () => {
+    return request(app)
+      .get("/api/articles/3")
+      .expect(200)
+      .then(({ body }) => {
+        const { articles } = body;
+        expect(articles).toMatchObject({
+          author: expect.any(String),
+          title: expect.any(String),
+          article_id: expect.any(Number),
+          body: expect.any(String),
+          topic: expect.any(String),
+          created_at: expect.any(String),
+          votes: expect.any(Number),
+          article_img_url: expect.any(String),
+        });
       });
-    
+  });
 
   test("404: Not Found when an id is valid but does not exist", () => {
     return request(app)
-    .get('/api/articles/500')
-    .expect(404)
-    .then((response) => {
-        expect(response.body.msg).toBe("Not Found")
-    })
-  })
+      .get("/api/articles/500")
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe("Not Found");
+      });
+  });
+});
+
+describe("GET /api/articles", () => {
+  test.only("200: return an array of all articles with all the correct properties", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then(({ body }) => {
+        const { articles } = body;
+        articles.forEach((article) => {
+          expect(article).toMatchObject({
+            author: expect.any(String),
+            title: expect.any(String),
+            article_id: expect.any(Number),
+            topic: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_img_url: expect.any(String),
+            comment_count: expect.any(String),
+          });
+        });
+      });
+  });
+
+  test.only("200: returns articles array sorted by date in descending order ", () => {
+    return request(app)
+      .get("/api/articles?sort_by=created_at&order=desc")
+      .expect(200)
+      .then(({body}) => {
+        const {articles} = body
+        expect(articles.length).toBe(13)
+        expect(articles).toBeSortedBy("created_at", {coerce: true, descending: true});
+      });
+  });
+
+  test.only("400: Bad Request when sort by query is not valid", () => {
+    return request(app)
+      .get("/api/articles?sort_by=cheese&order=desc")
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("Invalid Sort Request");
+      });
+  });
+
+  test.only("400: Bad Request when order by query is not valid", () => {
+    return request(app)
+      .get("/api/articles?sort_by=created_at&order=cheese")
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("Invalid Order Request");
+      });
+  });
```

Comment:
Really good error handling but this is not complete yet and is outside of the scope of this task.